doc/admin-guide/plugins/certifier.en.rst

```diff
@@ -0,0 +1,63 @@
+
+When the plugin sees an incoming HTTPS request, it does:
+1) Look up the SNI in `SslLRUList` and set up the context if a valid context exists. Otherwise, it will schedule a thread to retrieve such context from disk (or generate). If such a thread is already scheduled, it will put this SSL connection onto the queue.
+2) The retriever thread will first try to load the cert from disk. If no such certs exist and dynamic generation is enabled, the thread will generate a Certificate Signing Request (CSR) with provided SNI and sign it with root CA passed in from config options. The signed certificate is then written to the disk and all queued up SSL connections are set up with correct context and enabled.
+
+Setup
+=====
+* Signing cert, key, and serial
+   * Cert and key: For example, ``openssl req -newkey rsa:2048 -nodes -keyout ca.key -x509 -days 365 -out ca.cert``. Note that the cert/key should be generated without a challenge password.
+   * Serial: A text file containing a valid integer with a trailing new line.
+
+Plugin Configuration
+====================
+
+* Specify certificate generation related files. If any of the following parameters is missing, the dynamic generation will be disabled.
+   * ``--sign-cert=<path_to_certificate>`` (`optional`, default:empty/unused) - specifies the path to the root CA certficate. In most cases, this would be a self-signed certificate that is configured to be trusted by all potential clients. Path should be the path and file name of the cert. If it is relative, it is relative to the Traffic Server configuration directory.
+   * ``--sign-key=<path_to_key>`` (`optional`, default:empty/unused) - specifies the path to the root CA private key. In most cases, this would be generated alongside the self-signed root CA certificate.
+   * ``--sign-serial=<path_to_serial>`` (`optional`, default:empty/unused) - specifies the path to the serial number file. This will be used to assign serial numbers to certificates and keep all generated ones in sync. Serial file should be a number with a trailing newline.
+
+
+* Specify the certificates management related settings.
+   * ``--store=<path_to_certs_dir>`` (`required`, default:empty) - specifies the directory to use as the root of file system certificates storage.
+   * ``--max=<N>`` (`required`, default:empty) - specifies the upper limit on number of files kept in memory.
+
+
+Example Usage
+=============
+One use case would be routing incoming CONNECT request to another port on traffic server. With the certifier generating a trusted certificate, other plugins can act with a similar behavior to Man-In-The-Middle (logging interesting data for example).
```

Comment:
This needs more depth in terms of operational set up. A plantuml sequence diagram would be nice.